catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +171,67 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService("dog_legs.sch", "dog_paws.sch");
+        // test warnings
+        try {
+            service.validate(getMetacard("dog_4leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String warning : ex.getWarnings()) {
+                assertThat(warning.equals(service.sanitize(warning)), is(true));
+            }
+        }
+        // test errors
+        try {
+            service.validate(getMetacard("dog_3leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String error : ex.getErrors()) {
+                assertThat(error.equals(service.sanitize(error)), is(true));
+            }
+        }
+        // sanitize is called twice for each validate call
+        verify(service, times(4)).sanitize(anyString());
+    }
+
+    @Test
+    public void testSanitization() throws SchematronInitializationException {
+        SchematronValidationService service = getService(false,
```

Comment:
I've heard different opinions when it comes to unit testing with multiple assertions. Some like to have each assertion (or each test case) in it's own test so you can more easily see which assertion failed. But these are all testing the same regex, and it kinda makes sense to have it all in one method. Regardless, if you do break this into multiple methods (again) i think you should do the configuring of the service in the `@BeforeExam` so that each test would only be an assertion
@lessarderic, maybe you can lend some insight here about best practices? also @shaundmorris I think you commented on one of my PRs about this once